File: src/Api/Model/Languageforge/Lexicon/Config/LexTaskSemdom.php
<?php

namespace Api\Model\Languageforge\Lexicon\Config;

use Api\Model\Shared\Mapper\MapOf;

class LexTaskSemdom extends LexTask
{
    public function __construct()
    {
        $this->language = "en";
        $this->visibleFields = new MapOf();
        $this->type = LexTask::SEMDOM;

        // default values
        $this->visibleFields["definition"] = true;
        $this->visibleFields["partOfSpeech"] = true;
        $this->visibleFields["example"] = true;
        $this->visibleFields["translation"] = true;
        parent::__construct();
    }

    /** @var string */
    public $language;

    /** @var MapOf<boolean> */
    public $visibleFields;
}

File: src/netcore-api/SIL.XForge.WebApi.Server/Models/Lexicon/LexTaskSemdom.cs
using System.Collections.Generic;

namespace SIL.XForge.WebApi.Server.Models.Lexicon
{
    public class LexTaskSemdom : LexTask
    {
        public LexTaskSemdom()
        {
            Type = Semdom;

            VisibleFields["definition"] = true;
            VisibleFields["partOfSpeech"] = true;
            VisibleFields["example"] = true;
            VisibleFields["translation"] = true;
        }

        public string Language { get; set; } = "en";
        public Dictionary<string, bool> VisibleFields { get; protected set; } = new Dictionary<string, bool>();
    }
}
